src/bika/lims/browser/worksheet/views/analyses.py

```diff
@@ -123,15 +125,42 @@ def __init__(self, context, request):
                 "type": "remarks"
             }
 
+        self.set_analysis_remarks_modal = {
+            "id": "modal_set_analysis_remarks",
+            "title": _("Set remarks"),
+            "url": "{}/set_analysis_remarks_modal".format(
+                api.get_url(self.context)),
+            "css_class": "btn btn-outline-secondary",
+            "help": _("Set remarks for selected analyses")
+        }
+
         self.review_states = [
             {
                 "id": "default",
                 "title": _("All"),
                 "contentFilter": {},
+                "custom_transitions": [],
                 "columns": self.columns.keys(),
             },
         ]
 
+    def before_render(self):
+        super(AnalysesView, self).before_render()
+
+        if self.show_analysis_remarks_transition():
+            for state in self.review_states:
+                state["custom_transitions"] = [self.set_analysis_remarks_modal]
+
+    def show_analysis_remarks_transition(self):
+        """Check if the analysis remarks transitions should be rendered
+
+        XXX: Convert maybe better to a real WF transition with a guard
```

Comment:
Agree, a real transition would be better, so you won't need to bother about rendering this transition in listings: the button will be rendered as long as the transition is possible for all selected items

Reply:
Yes, that's the point: " the button will be rendered as long as the transition is possible for all selected items".
The advantage with the current approach is that you can simply select all, so I'm not sure if adding it as a real WF transition gives so much benefit...
However, let me think about it, I will add it then eventually in another PR.

Reply:
I know that adding a transition for this might seem a bit of overhead at a first glance. However, the benefits you get afterwards are worth, starting from the fact that you do no longer need to have "custom" code for this action anymore at the same time you get full-fledged permissions functionality